include/irr/builtin/glsl/bxdf/brdf/specular/beckmann_smith.glsl

```diff
@@ -1,38 +1,22 @@
 #ifndef _IRR_BSDF_BRDF_SPECULAR_BECKMANN_SMITH_INCLUDED_
 #define _IRR_BSDF_BRDF_SPECULAR_BECKMANN_SMITH_INCLUDED_
 
-#include <irr/builtin/glsl/bxdf/common.glsl>
+#include <irr/builtin/glsl/bxdf/common_samples.glsl>
 #include <irr/builtin/glsl/bxdf/brdf/specular/ndf/beckmann.glsl>
 #include <irr/builtin/glsl/bxdf/brdf/specular/geom/smith.glsl>
 #include <irr/builtin/glsl/bxdf/brdf/specular/fresnel/fresnel.glsl>
 #include <irr/builtin/glsl/math/functions.glsl>
 
 #include <irr/builtin/glsl/math/functions.glsl>
 
-//i wonder where i got irr_glsl_ggx_smith_height_correlated() from because it looks very different from 1/(1+L_v+L_l) form
-float irr_glsl_beckmann_smith_height_correlated(in float NdotV2, in float NdotL2, in float a2)
-{
-    float c2 = irr_glsl_smith_beckmann_C2(NdotV2, a2);
-    float L_v = irr_glsl_smith_beckmann_Lambda(c2);
-    c2 = irr_glsl_smith_beckmann_C2(NdotL2, a2);
-    float L_l = irr_glsl_smith_beckmann_Lambda(c2);
-    return 1.0 / (1.0 + L_v + L_l);
-}
-float irr_glsl_beckmann_smith_height_correlated_lambdaV(in float lambda_v, in float NdotL2, in float a2)
-{
-    float c2 = irr_glsl_smith_beckmann_C2(NdotL2, a2);
-    float L_l = irr_glsl_smith_beckmann_Lambda(c2);
-    return 1.0 / (1.0 + lambda_v + L_l);
-}
-
 irr_glsl_BSDFSample irr_glsl_beckmann_smith_cos_generate(in irr_glsl_AnisotropicViewSurfaceInteraction interaction, in vec2 _sample, in float ax, in float ay)
 {
     vec2 u = _sample;
     
     mat3 m = irr_glsl_getTangentFrame(interaction);
 
     vec3 localV = interaction.isotropic.V.dir;
-    localV = normalize(V*m);//transform to tangent space
+    localV = normalize(localV*m);//transform to tangent space
```

Comment:
your tanget frame shall be orthonormal, localV should also be normalized! so no need for normalization.